services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -1174,6 +1182,70 @@ private Integer fetchSystemStoreSchemaId(String clusterName, String storeName, S
     return response.getId();
   }
 
+  static boolean isPushJobFailedUserError(PushJobDetailsStatus status, PushJobCheckpoints pushJobCheckpoint) {
+    if (PushJobDetailsStatus.isFailed(status)) {
+      if (pushJobCheckpoint != null) {
+        switch (pushJobCheckpoint) {
+          case QUOTA_EXCEEDED:
+          case WRITE_ACL_FAILED:
+          case DUP_KEY_WITH_DIFF_VALUE:
+          case INPUT_DATA_SCHEMA_VALIDATION_FAILED:
+          case EXTENDED_INPUT_DATA_SCHEMA_VALIDATION_FAILED:
+          case RECORD_TOO_LARGE_FAILED:
+          case CONCURRENT_BATCH_PUSH:
+          case DATASET_CHANGED:
+          case INVALID_INPUT_FILE:
+          case DVC_INGESTION_ERROR_DISK_FULL:
+          case DVC_INGESTION_ERROR_MEMORY_LIMIT_REACHED:
+            return true;
+
+          default:
+            return false;
+        }
+      }
+    }
+    return false;
+  }
```

Comment:
Instead of hardcoding this decision, can we take this as a config, so it can be overridden based on how people operate VPJ? It might not be the same as how things are done at LI

Reply:
Good suggestion. Added a config for VPJ